core/src/event_id.rs

```diff
@@ -0,0 +1,66 @@
+use core::fmt::{Display, Formatter};
+
+use miden_crypto::hash::blake::Blake3_256;
+
+use crate::Felt;
+
+/// A type-safe wrapper around a [`Felt`] that represents an event identifier.
+///
+/// Event IDs are used to identify events that can be emitted by the VM or handled by the host.
+/// This newtype provides type safety and ensures that event IDs are not accidentally confused
+/// with other [`Felt`] values.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct EventID(Felt);
+
+impl EventID {
+    /// Computes the canonical event identifier for the given `name`.
+    ///
+    /// This function provides a stable, deterministic mapping from human-readable event names
+    /// to field elements that can be used as event identifiers in the VM. The mapping works by:
+    /// 1. Computing the BLAKE3 hash of the event name (produces 32 bytes)
+    /// 2. Taking the first 8 bytes of the hash
+    /// 3. Interpreting these bytes as a little-endian u64
+    /// 4. Reducing modulo the field prime to create a valid Felt
+    ///
+    /// This ensures that identical event names always produce the same event ID, while
+    /// providing good distribution properties to minimize collisions between different names.
+    pub fn from_name(name: impl AsRef<str>) -> Self {
+        let digest_bytes: [u8; 32] = Blake3_256::hash(name.as_ref().as_bytes()).into();
+        let event_bytes: [u8; 8] = digest_bytes[0..8].try_into().unwrap();
+
+        Self(Felt::new(u64::from_le_bytes(event_bytes)))
```

Comment:
Should we use `hash_string_to_word()` here as a building block? Then, this could look like:
```rust
Self(hash_string_to_word(name.as_ref())[0])
```